README.md

```diff
@@ -1,57 +1,253 @@
+```
+
+Now let's add in a ```create``` action in the posts' controller and have it grab the params, store them in an instance variable and then redirect to the new page:
+
+```ruby
+def create
+  session[:form_params] = params.inspect
+  redirect_to new_post_path
+end
+```
+
+I'm storing the values from the form in the session so when we redirect back to the form we can print these values out on the page to ensure the params are processing properly. Lastly, let's update the ```new.html.erb``` view template so it prints out the form values:
+
+```ERB
+<%= render "form" %>
+
+<%= session[:form_params] if session[:form_params] %>
+```
+
+This is temporary code that we will use to make sure our form is working properly, I'm storing the values in the session so it's more straightforward to test and pass between the controller actions. If you run the rails server and go to the ```posts/new``` page and fill in the title and description form elements and click submit you will find a new type of error:
+
+![InvalidAuthenticityToken](http://reif.io/lib/flatiron/InvalidAuthenticityToken.png)
+
+Which leads us to a very important part of Rails forms: CSRF.
+
+
+## What is CSRF?
+
+First and foremost, CSRF is an acronym for: Cross-Site Request Forgery (CSRF). Instead of giving a boring explanation of what happens during a CSRF request, let's walk through a real life example of a Cross-Site Request Forgery hack:
+
+1. You go to your bank website and login, you check your balance and then open up a new tab in the browser and go to your favorite meme site.
+
+2. Without you knowing the meme site is actually a hacking site that has scripts running the background as soon as you land on the page.
+
+3. One of the scripts on the site hijacks the banking session that you have open in the other browser and submits a form request to transfer money to their account.
+
+4. The banking form can't tell that the form request wasn't made by you, so it all goes through just like you were the one who made the request.
+
+This is a Cross-Site Request Forgery request, one site makes a request to another site via a form. Rails blocks this from happening by default by requiring that a unique authenticity token is submitted with each form. This authenticity token is stored in the session and can't be hijacked by hackers since it performs a match check when the form is submitted and will throw an error if the token isn't there or doesn't match.
+
+To fix this ```ActionController::InvalidAuthenticityToken``` error, we can integrate the ```form_authenticity_token``` helper into the form as a hidden field:
+
+```ERB
+<h3>Post Form</h3>
+
+<form action="<%= posts_path %>" method="post">
+  <label>Post title:</label><br>
+  <input type="text" id="post_title" name="post[title]"><br>
+
+  <label>Post Description</label><br>
+  <textarea id="post_description" name="post[description]"></textarea><br>
+
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
```

Comment:
Sorry, I think I didn't explain myself well. the `hidden_field_tag` here should be replaced with a normal hidden field